Dockerfile.bootstrap-provider

```diff
@@ -8,28 +8,12 @@ ENV USER_UID=1001 \
 COPY --chown=${USER_UID} . /workspace
```

Comment:
Although it's a simpler approach, this would make caching more difficult, and also would require better care with the `.dockerignore` file and in general it's adding potential unnecessary files (if not in the .dockerignore) into the image.
What are the motivations for this change? Just simplifying or is there anything else behind it?

Reply:
I was just copying from [how assisted-service was doing it](https://github.com/openshift/assisted-service/blob/36feaa473765645c1b7a19e18573365c7d255175/Dockerfile.assisted-service-rhel9-mce#L8-L11) to try to get something to work.

Now that we have it passing I can try to roll back some of these.

Reply:
I think ubi golang images expect us to work in `/opt/app-root/src` path, which already has all permission necessary for the default user. We can change it to something else if need be though